bin/quipucords-installer

```diff
@@ -2,60 +2,144 @@
 #
 # Quipucords Installer
 #
-export VERBOSE="n"
-INSTALLER_COMMAND="quipucords-installer"
-QUIPUCORDS_PKG_ROOT="${QUIPUCORDS_PKG_ROOT:-/usr/share/${INSTALLER_COMMAND}}"
-QUIPUCORDS_BIN_DIR=${QUIPUCORDS_PKG_ROOT}/bin
+set -e
 
-CONFIG_PATH="${QUIPUCORDS_PKG_ROOT}/config"
-ENV_PATH="${QUIPUCORDS_PKG_ROOT}/env"
+if [ "$(id -u)" -eq 0 ]; then
+  echo "$0 must not run as root." 1>&2
+  exit 1
+fi
 
-QUIPUCORDS_USER_PATH="${HOME}/.local/share/quipucords"
-ENV_USER_PATH="${HOME}/.config/quipucords/env/"
-SYSTEMD_USER_PATH="${HOME}/.config/containers/systemd/"
+usage() {
+  echo "Usage: $0 [-v|--verbose] [--override-conf-dir value] <command>"
+  echo "Options:"
+  echo "  -v, --verbose                 Enable verbose output mode"
+  echo "  --override-conf-dir value     Path of a directory containing config overrides"
+  echo "Commands:"
+  echo "  install                       Install Quipucords"
+  echo "  uninstall                     Uninstall Quipucords"
+  echo "  create-server-password        Create a Quipucords server password"
+  echo "  create-app-secret             Create a Quipucords application secret"
+  exit 1
+}
 
-if [ "${1}" == "-v" ] || [ "${1}" == "--verbose" ]; then
-  VERBOSE="y"
-  shift
-fi
+check_prereqs() {
+  local requirements="base64 basename cp diff echo grep head mkdir podman tr read rm sed systemctl"
+  for requirement in $requirements; do
+    if ! command -v "$requirement" >/dev/null 2>&1; then
+      echo "Error: $requirement is not installed or is not in the PATH." >&2
+      exit 1
+    fi
+  done
+}
 
-COMMAND="${1}"
-shift
+set_default_vars() {
+  VERBOSE="n"
+  OVERRIDE_CONF_DIR="" # Optional
 
-function debug_msg {
-  if [ "${VERBOSE}" == "y" ]; then
-    echo "$@"
+  # All external programs quipucords-installer needs are in standard bin paths.
+  # Do not allow a user's custom PATH environment to interfere.
+  export PATH="/usr/bin:/bin:/usr/sbin:/sbin"
+
+  INSTALLER_BIN_DIR=$(dirname "$(realpath "$0")")
```

Comment:
Not that it really matters, but `$0` will fail if script is sourced instead of executed.

Since we require bash, we could use `$(realpath ${BASH_SOURCE[0]})` instead, which will point on to current script no matter if it is being executed or sourced.

Reply:
I would rather block sourcing the script since that could introduce unexpected changes to their running shell environment that we do not intend to support. We do `export` a few variables that should not leak into the user's environment. (Whether they should be `export`ed at all is another matter…)

To enforce that, I added checks at the top of each script like this:
```bash
if [[ "${BASH_SOURCE[0]}" != "${0}" ]]; then
  echo "This script is intended to be run, not sourced." >&2
  return 1
fi
```

Reply:
That works too :)